File: lib/features/auth/models/product.dart
import 'dart:convert';

class Product {
  final String name;
  final String description;
  final double price;
  final double quantity;
  final String category;
  final List<String> images;
  final String? id;
  final String? userId;
  Product({
    required this.name,
    required this.description,
    required this.price,
    required this.quantity,
    required this.category,
    required this.images,
    this.id,
    this.userId,
  });

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'name': name,
      'description': description,
      'price': price,
      'quantity': quantity,
      'category': category,
      'images': images,
      'id': id,
      'userId': userId,
    };
  }

  factory Product.fromMap(Map<String, dynamic> map) {
    return Product(
      name: map['name'] as String,
      description: map['description'] as String,
      price: map['price'] as double,
      quantity: map['quantity'] as double,
      category: map['category'] as String,
      images: List<String>.from(map['images'] as List<String>),
      id: map['_id'],
      userId: map['userId'],
    );
  }

  String toJson() => json.encode(toMap());

  factory Product.fromJson(String source) =>
      Product.fromMap(json.decode(source) as Map<String, dynamic>);
}

File: server/model/product.js
const mongoose =require('mongoose');

const productSchema =mongoose.Schema({
    
        name:{
            required: true,
            trim: true,
            type: String
        },
        description:{
            required: true,
            trim: true,
            type: String
        },
        price:{
            required: true,
            type: Number
        },
        quantity:{
            required: true,
            type: Number
        },
        category:{
            required: true,
            type: String
        },
        images:[
            {
            required: true,
            type: String
        },
    ],
    
    
    }
    );
const Product =mongoose.model('Product',productSchema);
module.exports=Product;